python/lsst/rapid/analysis/mountTorques.py

```diff
@@ -0,0 +1,186 @@
+import time
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.time import Time
+import asyncio
+
+try:
+    from lsst_efd_client import merge_packed_time_series as mpts
+    HAS_EFD_CLIENT = True
+except ImportError:
+    HAS_EFD_CLIENT = False
+
+GOOD_IMAGE_TYPES = ['OBJECT', 'SKYEXP', 'ENGTEST']
+
+
+def getData(client, dataSeries, startTime, endTime):
+    """A synchronous get of the data from the EFD
+
+    This exists so that the top level functions don't all have to be async def
+    """
+    loop = asyncio.get_event_loop()
+    return loop.run_until_complete(client.select_time_series(dataSeries, ['*'], startTime, endTime))
+
+
+def plotMountTracking(dataId, butler, client, figure, saveFilename, butlerGeneration, logger):
+    """Queries EFD for given exposure and checks if there were tracking errors.
+        Parameters
+        ----------
+        XXX Write the docs
+
+        Returns
+        -------
+        None
+    """
+    # lsst-efd-client is not a required import at the top here, but is
+    # implicitly required as a client is passed into this function so is not
+    # rechecked here.
+
+    start = time.time()
+
+    if butlerGeneration == 3:
+        expId = dataId['expId']
+        mData = butler.get('raw.metadata', detector=0, exposure=expId)
+        dataIdString = str(expId)
+    else:
+        mData = butler.get('raw_md', **dataId)
+        dataIdString = f"{dataId['dayObs']}-{dataId['seqNum']}"
+
+    imgType = mData['IMGTYPE']
+    tStart = mData['DATE-BEG']
+    tEnd = mData['DATE-END']
+    elevation = mData['ELSTART']
+    azimuth = mData['AZSTART']
+    exptime = mData['EXPTIME']
+    logger.debug(f"dataId={dataIdString}, imgType={imgType}, Times={tStart}, {tEnd}")
+
+    if imgType not in GOOD_IMAGE_TYPES:
+        logger.info(f'Skipping image type {imgType}')
+        return False
+    if exptime < 1.99:
+        logger.info('Skipping sub 2s expsoure')
+        return False
+
+    end = time.time()
+    elapsed = end-start
+    logger.debug(f"Elapsed time for butler query = {elapsed}")
+
+    start = time.time()
+    # Time base in the EFD is still a big mess.  Although these times are in
+    # UTC, it is necessary to tell the code they are in TAI. Then it is
+    # necessary to tell the merge_packed_time_series to use UTC.
+    # After doing all of this, there is still a 2 second offset,
+    # which is discussed in JIRA ticket DM-29243, but not understood.
+
+    t_start = Time(tStart, scale='tai')
+    t_end = Time(tEnd, scale='tai')
+    logger.debug(f"Tstart = {t_start.isot}, Tend = {t_end.isot}")
+
+    mount_position = getData(client, "lsst.sal.ATMCS.mount_AzEl_Encoders", t_start, t_end)
```

Comment:
The calling code is already doing that, and there's a comment about this exact thing on L56-58. The protected import at the top is just to stop this file breaking the module for when that's not installed, but the function in question has the client passed in, so can't really not have it at runtime, so it should be a non-issue. The protected import at the top doesn't really need to be setting the `HAS_EFD_CLIENT` bool tbh, but it just seemed like good practice to stick that in up there as part of the pattern in case it's useful elsewhere in future.

If you think that's a bit odd then I can remove that bool, but I don't think checking on it within the function is necessary, given that the caller is responsible for passing the `client` in.

Reply:
If you're not using it, you should get rid of it.  The `mpts` calls below might be better protected (I just wrote a trivial example that shows that I think shows I can pass a non-client to `getData` and have it tell me it succeeded).